File: python/7_kyu/AlphabeticalAddition.py
# Your task is to add up letters to one letter.

# The function will be given a variable amount of arguments, each one being a letter to add.
# Notes:

#     Letters will always be lowercase.
#     Letters can overflow (see second to last example of the description)
#     If no letters are given, the function should return 'z'

# Examples:

# add_letters('a', 'b', 'c') = 'f'
# add_letters('a', 'b') = 'c'
# add_letters('z') = 'z'
# add_letters('z', 'a') = 'a'
# add_letters('y', 'c', 'b') = 'd' # notice the letters overflowing
# add_letters() = 'z'

def add_letters(*letters):
    alphabet = "zabcdefghijklmnopqrstuvwxy"
    result=sum(alphabet.find(x) for x in letters ) 
    if result < 1 :
        return 'z'
    else :
        return alphabet[result%26]

print(add_letters('a', 'b', 'c'))
print(add_letters('a', 'b') )
print(add_letters('z') )
print(add_letters('y', 'c', 'b'))
print(add_letters())

File: java/7_kyu/AlphabeticalAddition.java
/**
Your task is to add up letters to one letter.

The function will be given an array of single character Strings, each one being a letter to add.
Notes:

    Letters will always be lowercase.
    Letters can overflow (see second to last example of the description)
    If no letters are given, the function should return 'z'

Examples:

addLetters("a", "b", "c") = "f"
addLetters("a", "b") = "c"
addLetters("z") = "z"
addLetters("z", "a") = "a"
addLetters("y", "c", "b") = "d" // notice the letters overflowing
addLetters() = "z"
 * 
 */

public class AlphabeticalAddition {

    public static void main(String[] args) {
        System.out.println(addLetters("a","b","c"));
        System.out.println(addLetters("a","b"));
        System.out.println(addLetters("z"));
        System.out.println(addLetters("z", "a"));
        System.out.println(addLetters("y", "c", "b"));
        System.out.println(addLetters());
    }


    public static String addLetters(String... letters) {
        int current = 0;
        for( String c : letters){
            current += (int)c.charAt(0) - 96;
        }
        current = current % 26 + 96;
        if( current == 96 ){
            return "z";
        }

        return  String.valueOf((char)current) ;
    }

}
